### PR TITLE
Binance extraction

### PR DESCRIPTION
**Pull request by AdalyatNazirov**
#### Description

Extract Binance to separate repository

#### Related Issue

N/A

#### Motivation and Context

We want to make Lean project modular and lightweight

#### Requires Documentation Change

N/A

#### How Has This Been Tested?

- manually
  - trading
  - DQH
  - ToolBox
 
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Tests/Brokerages/Binance/BinanceFeeModelTests.cs [hunk 1]
```diff
@@ -1,234 +0,0 @@
...
-using QuantConnect.Securities;
-using System;
-
-namespace QuantConnect.Tests.Brokerages.Binance
-{
-    [TestFixture]
-    public class BinanceFeeModelTests
-    {
-        [Test]
-        public static void GetFeeModelTest()
```
Comment (Martin-Molinero) [inline Tests/Brokerages/Binance/BinanceFeeModelTests.cs:31]: The fee/brokerage model are still remaining? Should keep their tests together 👌

**Review by Martin-Molinero — APPROVED** [review]

**Comment by Zorgino** [comment]
Can someone point out to me where it got extracted?
This kind of change should be considered breaking as, apparently, LEAN does not support a feature of Binance trading anymore.

**Comment by Martin-Molinero** [comment]
> Can someone point out to me to where it got extracted? This kind of change should be considered breaking as, apparently, LEAN does not support a feature of Binance trading anymore.

Hey @Zorgino, sorry about the confusion here, Lean still supports Binance trading, locally through the [CLI](https://www.lean.io/cli/) and in the [cloud ](https://www.quantconnect.com/) too. We are moving Lean into a more plugin approach where it can scale better.

**Comment by Zorgino** [comment]
> > Can someone point out to me to where it got extracted? This kind of change should be considered breaking as, apparently, LEAN does not support a feature of Binance trading anymore.
> 
> Hey @Zorgino, sorry about the confusion here, Lean still supports Binance trading, locally through the [CLI](https://www.lean.io/cli/) and in the [cloud ](https://www.quantconnect.com/) too. We are moving Lean into a more plugin approach where it can scale better.

Thanks, @Martin-Molinero. I would like to use it locally using C# though.
I suppose Binance brokerage is going to be implemented as [a brokerage templates](https://github.com/QuantConnect/Lean.Brokerages.Template), right?
Frankly, it should've been done prior to removing it from Lean platform.

Is there any WIP regarding this?

**Pull request merged**